.github/workflows/namerank-api-lambda-deploy.yml

```diff
@@ -0,0 +1,138 @@
+name: NameRank API - Lambda Deploy
+
+on:
+  push:
+    branches:
+      - main
+      - staging
+    paths:
+      - "packages/namerank-python/**"
+      - "apps/api.namerank.io/**"
+      - ".github/workflows/namerank-api-lambda-deploy.yml"
+  workflow_dispatch:
+
+permissions:
+  # `id-token: write` is required for the auth to AWS
+  id-token: write
+  contents: read
+
+concurrency:
+  group: ${{ github.workflow }}
+  # `cancel-in-progress: false` creates a queue for workflow runs, such that
+  # only one instance of this workflow runs at a time.
+  cancel-in-progress: false
+
+jobs:
+  build-image-deploy:
+    name: Build and deploy NameRank API Lambda
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout this repo
+        uses: actions/checkout@v4
+
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v4
+        with:
+          role-to-assume: ${{ secrets.AWS_ROLE}}
+          aws-region: ${{ secrets.AWS_REGION }}
+      
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v3
+        with:
+          platforms: arm64
+      
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+        
+      - name: Setup Terraform
+        uses: hashicorp/setup-terraform@v2
+        with:
+          terraform_version: "1.5.7"
+          terraform_wrapper: false
+    
+
+      - name: Build and deploy lambda
+        env:
+          PROVIDER_URI_MAINNET: ${{ secrets.PROVIDER_URI_MAINNET }}
+          PROVIDER_URI_SEPOLIA: ${{ secrets.PROVIDER_URI_SEPOLIA }}
+          ALCHEMY_URI_MAINNET: ${{ secrets.ALCHEMY_URI_MAINNET }}
+          ALCHEMY_URI_SEPOLIA: ${{ secrets.ALCHEMY_URI_SEPOLIA }}
+          ENS_SUBGRAPH_URL_MAINNET: ${{ secrets.ENS_SUBGRAPH_URL_MAINNET }}
+          ENS_SUBGRAPH_URL_SEPOLIA: ${{ secrets.ENS_SUBGRAPH_URL_SEPOLIA }}
+          AWS_REGION: ${{ secrets.AWS_REGION }}
+          CERTIFICATE_NAME: ${{ secrets.CERTIFICATE_NAME }}
+          HOSTED_ZONE_NAME: ${{ secrets.HOSTED_ZONE_NAME }}
+        run: |
+          if [[ ${{ github.ref }} == 'refs/heads/main' ]]; then
+          STAGE="prod"
+          DOMAIN_NAME=${{ secrets.PROD_DOMAIN_NAME }}
```

Comment:
@BanaSeba all these files are moved from namerank repo. I guess `secrets.PROD_DOMAIN_NAME` and `secrets.STAGING_DOMAIN_NAME` should be changed?